### PR TITLE
web/app: remove hover on navbar items to stop focus from being moved

### PR DESCRIPTION
**Pull request by limitedmage**
The dropdown menu items in the global navbar (Search and, in App, Feedback) move focus when you hover over them. This is bizarre, ugly, and most likely a [WCAG violation](https://www.w3.org/TR/WCAG21/#focus-order). I spent some time yesterday and the root cause is that the [Reach menu button we use does not support hover at all](https://github.com/reach/reach-ui/issues/278) and we're hacking it by [sending a mousedown event when you hover over the button](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/client/web/src/nav/NavBar/NavDropdown.tsx?L58).

There are two options I can think of to mitigate this bug:
1. Get rid of hover events and only open the dropdowns on click. This is the easiest option, but obviously removes the ease of accessing the menu for mouse users.
2. Rewrite the dropdown nav items to use a custom menu instead of Reach. This is more expensive because we have to reimplement lots of a11y stuff here, but we can have more fine-grained control of different UI flows for mouse hover vs mouse click vs keyboard users.

I went with the first option following @almeidapaulooliveira's [feedback](https://sourcegraph.slack.com/archives/C0HMGV90V/p1683735888755969?thread_ts=1683734992.157499&cid=C0HMGV90V). 

The following changes were applied to `NavDropdown`:

- Removed all hover and touch events; now only a click will open the menu.
- Removed the split button; now clicking anywhere on the button will always open the menu.
- Made the mobile home item into a "generic" home item and always visible when present.
- Made the home item optional (since the Feedback menu item in the desktop app doesn't have a home item/route).
- Added customizable a11y names; in the past, the only dropdown was the Search one, but in App we now have Feedback; the a11y labels were still saying "Search" before this change.

Additionally, the following changes were applied for polish:

- The double-focus ring on the back/forward buttons in the Tauri app has been fixed
- Styling changes made to simplify code

## Test plan

The global navbar has extensive visual and behavior test coverage.

### CODE REVIEW

**Comment by sourcegraph-buildkite** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| -0.05% (-1.36 kb) | -0.01% (-1.36 kb) | 0.00% (0.00 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/jp/navbarfocus/compare-report.html;tab=live_object) for a full comparison between the commits b6d955a and 5072054 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Review by slimsag — APPROVED** [review]
I'm not qualified to review this, just giving general approval :)

File: client/web/src/app/TauriNavigation.module.scss [hunk 1]
```diff
@@ -6,6 +6,10 @@
 .button {
     padding: 0 0.25rem;
 
+    &:focus-visible:not(:disabled):not(.disabled):not([aria-disabled='true']) {
```
Comment (vovakulikov) [inline client/web/src/app/TauriNavigation.module.scss:9]: Thanks for `[aria-disabled='true']`!

File: client/web/src/nav/NavBar/NavDropdown.tsx [hunk 2]
```diff
@@ -13,7 +13,7 @@ import navItemStyles from './NavItem.module.scss'
 
 export interface NavDropdownItem {
     content: React.ReactNode | string
-    path: string
+    path: string // To match against the current path to determine if the item is active
```
Comment (vovakulikov) [inline client/web/src/nav/NavBar/NavDropdown.tsx:16]: Super minor, I think that we usually do JSDOC /** annotation if we want to describe some prop or property 
Just comments // don't show up in IDE/VSCode tootlips

File: client/web/src/nav/NavBar/NavDropdown.tsx [hunk 3]
```diff
@@ -120,70 +73,35 @@ export const NavDropdown: React.FunctionComponent<React.PropsWithChildren<NavDro
...
-                                    key={toggleItem.path}
-                                    to={toggleItem.path}
-                                    className={styles.showOnTouchScreen}
-                                    index={-1}
-                                >
-                                    {mobileHomeItem.content}
-                                </MenuLink>
+                            <MenuList className={styles.navDropdownContainer} targetPadding={EMPTY_RECTANGLE}>
+                                {homeItem && (
+                                    <MenuLink as={Link} key={toggleItem.path} to={toggleItem.path} index={-1}>
```
Comment (vovakulikov) [inline client/web/src/nav/NavBar/NavDropdown.tsx:101]: Why do we need to make it index={-1}, I actually thought that this is a bug that the first item isn't accessible by keyboard

**Comment by limitedmage** [comment]
> * It doesn't have anything to this PR, but the mobile version of the menu overlaps with the search box input. We probably should put an isolation rule for scrollable content block and give header z-index value. (feel free to ignore this if it's not the case)
> 
> Screenshot 2023-05-11 at 14 12 12

@vovakulikov This is a pre-existing bug due to the new search box using `z-index: 1` (not sure why this is the case, @fkling any ideas here?). The *right* way to fix this is to have the mobile menu use the `MenuButton` Reach component, which relies on React Portals, but this is a bit complex; I'll try to get to it in a future PR. Without using portals, the monkey-patch fix for this is to add `z-index: 2` (or a higher number), but this is a ticking time bomb obviously. I'll add that in this PR as a temporary fix.

File: client/web/src/nav/NavBar/NavDropdown.tsx (hunk 3 above)
Comment (limitedmage) [reply client/web/src/nav/NavBar/NavDropdown.tsx:101]: Good call, this was pre-existing code, it's removed now.

File: client/web/src/nav/NavBar/NavDropdown.tsx (hunk 2 above)
Comment (limitedmage) [reply client/web/src/nav/NavBar/NavDropdown.tsx:16]: Good call! Modified all comments in this file to use JSDOC-style.

**Pull request merged**